File: code/python/src/rl_agent/pre_run.py
"""
A Reinforcement Learning Application for Statistical Arbitrage
ATRGStrategies

This module sets up the environment and imports all necessary dependencies
for the reinforcement learning application.
"""

from pathlib import Path

# Add required packages to requirements.txt:
# numpy
# pandas
# statsmodels
# scikit-learn
# matplotlib
# seaborn
# tqdm
# typing

# Set up data path
DATA_PATH = Path(__file__).parent.parent / "data" / "raw" / "1min"

# Import all required modules
from . import mcc  # renamed from MCC.py to follow Python naming conventions
from . import (
    attributes,
    cointegration,
    discretization,
    financial_functions,
    getdata,
    q_learning,
    reward,
    rl_utils,
    state_space,
    tdva,
    utils,
)

# Make all modules available at package level
__all__ = [
    # "object_utils",
    "utils",
    "getdata",
    "cointegration",
    "attributes",
    "financial_functions",
    "discretization",
    "state_space",
    "reward",
    "mcc",
    "q_learning",
    "tdva",
    "rl_utils",
]

File: code/R/src/rl_agent/pre_run.R
# A Reinforcement Learning Application for Statistical Arbitrage
# ATRGStrategies

# -----------------------------------------------------------------
# import dependencies (files and libraries)
rm(list = ls())
#install.packages("plyr")
if (!'pacman' %in% installed.packages()) install.packages('pacman')
library(pacman)

p_load(
  gsubfn,
  plyr,
  purrr,
  purrrlyr,
  R6,
  gridExtra,
  glue,
  discretization,
  assertthat,
  dplyr,
  ggplot2,
  scales,
  colorspace,
  urca
)

# -----------------------------------------------------------------
#region Dependencies

#setwd("C:/Users/user/Desktop/ATRG/ROO/atrg-strategies/src/main/R")
#setwd("C:/Users/Krzysztof/Desktop/ab/ATRGStrategies/atrg-strategies/src/main/R")
data_path <- file.path(getwd(), 'data', '1min')
#data_path <- c("C:\\Users\\user\\Desktop\\ATRG\\data\\data\\")
# biger datahas to be kept outside project path
#
source("./src/rl_agent/object.R")
source("./src/rl_agent/utils.R")
source("./src/rl_agent/getdata.R")
source("./src/rl_agent/cointegration.R")
source("./src/rl_agent/attributes.R")
source("./src/rl_agent/financial_functions.R")
source("./src/rl_agent/discretization.R")
source("./src/rl_agent/state_space.R")
source("./src/rl_agent/reward.R")
source("./src/rl_agent/MCC.R")
source("./src/rl_agent/q_learning.R")
source("./src/rl_agent/tdva.R")
source("./src/rl_agent/rl_utils.R")
